File: src/shared/base/BaseRepository.ts
import { BaseEntity } from './BaseEntity';

export abstract class BaseRepository<T extends BaseEntity> {
  private readonly map = new Map<string, T>();
  protected getById(id: string): T {
    return this.map.get(id);
  }
  protected add(id: string, item: T) {
    this.map.set(id, item);
  }
  protected delete(id: string) {
    this.map.delete(id);
  }
  protected clear() {
    this.map.clear();
  }
}

File: src/main/java/com/feyz/checkoutcase/app/repository/BaseRepository.java
package com.feyz.checkoutcase.app.repository;

import java.util.HashMap;
import java.util.Map;


public abstract class BaseRepository<T> {

    private Map<String, T> map = new HashMap<String, T>();

    protected void add(String id, T item) {
        this.map.put(id, item);
    }
    protected void remove(String id) {
        this.map.remove(id);
    }

    protected T get(String id) {
        return this.map.get(id);
    }

    protected void clear() {
        this.map.clear();
    }
}
